doc/source/data/examples/index.rst

```diff
@@ -0,0 +1,51 @@
+.. _datasets-examples-ref:
+
+========
+Examples
+========
+
+.. tip:: Check out the Datasets :`User Guide <data_user_guide>` To learn more about
+  Datasets' features in-depth.
+
+.. _datasets-recipes:
+
+Basic Data Processing Examples
```

Comment:
I'm not sure if Basic v.s. Large-Scale a good categorization, e.g.  basic processing (think map_batches()) can also run at large-scale.

Reply:
Hmm yeah good point, "Basic" was supposed to connote both the simplicity of the data processing operations and the size of the data, but I think it only does the former. I'm still not sure what the best example groupings are.

Here are some example group ideas:
- Small-scale data processing - this should be simple data processing (one-liner transforms and basic aggregations) on small data
  - I'm focusing on these right now for tabular, text, and imagery data
- Large-scale data processing -  this should be simple data processing (one-liner transforms and basic aggregations) on large data and a multi-node cluster
  - Our large-scale ML ingest example is ~close to this, although it's very light on the data processing (it's just read + shuffle)
  - Do we want to delineate examples based on data/cluster scale?
- Simple/basic data processing - this should be simple data processing operations (one-liner transforms and basic aggregations)
  - Not sure if this is a good category; are we going to have a "complex data processing" group? I don't think so, since those "complex" examples will be tailored to use cases, so framing it as "complex" probably isn't right.
- End-to-end ML pipelines - this would be full ML ingest and/or batch inference example pipelines, with real preprocessing
  - These would be closer to the Ray Train examples, with a bit more preprocessing involved: https://docs.ray.io/en/master/train/examples.html#ray-datasets-integration-examples

Reply:
I'm going forward with these example groups, where the first two are added in this PR, and we'll add the third group once we have e2e ML pipeline examples.

Reply:
I don't have a good idea here (sorry for pointing out but without a solution). I think Basic Data Processing sounds a good category, which as you mentioned can showcase (and breakdown by) the capabilities of data types and core features. E2E ML pipelines sound good to me, as that's our target workload.